gax-java/gax/src/test/java/com/google/api/gax/core/GaxPropertiesTest.java

```diff
@@ -57,10 +57,85 @@ public void testGaxVersion() {
     }
   }
 
+  private static String originalJavaVersion = System.getProperty("java.version");
+  private static String originalJavaVendor = System.getProperty("java.vendor");
+  private static String originalJavaVendorVersion = System.getProperty("java.vendor.version");
+
+  public static void cleanup() {
+    if (Strings.isNullOrEmpty(originalJavaVersion)) {
+      System.clearProperty("java.version");
+    } else {
+      System.setProperty("java.version", originalJavaVersion);
+    }
+
+    if (Strings.isNullOrEmpty(originalJavaVendor)) {
+      System.clearProperty("java.vendor");
+    } else {
+      System.setProperty("java.vendor", originalJavaVendor);
+    }
+
+    if (Strings.isNullOrEmpty(originalJavaVendorVersion)) {
+      System.clearProperty("java.vendor.version");
+    } else {
+      System.setProperty("java.vendor.version", originalJavaVendorVersion);
+    }
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfoCaseGraalVM() {
+
+    System.setProperty("java.version", "17.0.3");
+    System.setProperty("java.vendor", "GraalVM Community");
+    System.setProperty("java.vendor.version", "GraalVM CE 22.1.0");
+
+    String runtimeInfo = GaxProperties.getRuntimeVersion();
+    assertEquals("17.0.3__GraalVM-Community__GraalVM-CE-22.1.0", runtimeInfo);
+    cleanup();
+  }
+
   @Test
-  public void testGetVersion_nativeImage() {
-    System.setProperty(PROPERTY_IMAGE_CODE_KEY, PROPERTY_IMAGE_CODE_VALUE_RUNTIME);
-    String javaVersion = GaxProperties.getJavaVersion();
-    assertTrue(javaVersion.endsWith("-graalvm"));
+  public void testGetJavaRuntimeInfoCaseOne() {
```

Comment:
Avoid nameing "CaseOne". It doesn't tell what it is testing.

Reply:
+1. We have a pretty good [internal guide](https://engdoc.corp.google.com/eng/doc/devguide/testing/unit/best-practices.md?cl=head&polyglot=java#naming) for unit test naming conventions. This could be something like `testGetRuntimeVersion_shouldIncludeVendorVersionIfItExists` or `testGetRuntimeVersion_vendorVersionExist_returnsWithVendorVersion`.